src/transport/SecureSession.cpp

```diff
@@ -144,12 +151,18 @@ void SecureSession::MarkInactive()
     case State::kDefunct:
         FALLTHROUGH;
     case State::kActive:
+        // Make sure we don't notify "deferred" that we're being removed.  This
+        // ensures that "deferred" is in fact in our holder list.
+        RemoveHolder(deferred);
```

Comment:
What happens if we pass a random holder that wasn't in our list (i.e. not pointing to this at all) in the first place?

Reply:
Then this will crash.  Since the only callsite for this should be the exchange context, that should not be an issue in practice....